metricbeat/module/windows/perfmon/pdh_windows.go

```diff
@@ -268,6 +267,17 @@ func (q *Query) AddCounter(counterPath string, format Format, instanceName strin
 	return nil
 }
 
+// matchInstanceName will check first for instance and then for any objects names
+func matchInstanceName(counterPath string, instanceName *string) {
```

Comment:
Hmm instanceName is a string in line 237 but now you are passing it as a pointer? Maybe this function can be refactored so it's more readable 😄

Reply:
changed it to a pure method instead, should look more readable